src/jit/codegenarm64.cpp

```diff
@@ -2420,23 +2420,27 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
             // Floating point divide never raises an exception
             genCodeForBinary(treeNode);     
         }
-        else  // integer divide operation
+        else  // an integer divide operation
         {
             GenTreePtr divisorOp = treeNode->gtGetOp2();
+            emitAttr   size      = EA_ATTR(genTypeSize(genActualType(treeNode->TypeGet())));
 
             // TODO-ARM64-CQ: Optimize a divide by power of 2 as we do for AMD64
             
             if (divisorOp->IsZero())
             {
-                emitJumpKind jmpEqual = genJumpKindForOper(GT_EQ, CK_SIGNED);
-                genJumpToThrowHlpBlk(jmpEqual, SCK_DIV_BY_ZERO);
-                // We don't need to generate the sdiv/udiv instruction
+                // We unconditionally throw a divide by zero exception
+                genJumpToThrowHlpBlk(EJ_jmp, SCK_DIV_BY_ZERO);
+
+                // We still need to produce a register result 
+                emit->emitIns_R_R(INS_mov, size, targetReg, REG_ZR);
```

Comment:
For correctness we have to call  genProduceReg(treeNode);
That method handles spilling and  marking something a a GC or non GC pointer.
 To me it make sense to put an updated value into the destination register rather than leave a possibly uninitialized value there.  The extra code size of one instruction for the case of an explicit division by the constant zero is a negligible cost to prevent a possible GC pointer information leak.